fbot_world_bringup/nodes/pose.py

```diff
@@ -97,14 +98,41 @@ def setStaticPose(self):
     '''
     @brief Stores all target poses as static entries in the Redis database.
     '''
-    poses = self.targets
     with self.r.pipeline() as pipe:
-      for p_id, pose in poses.items():
-        key = 'target/' + p_id + '/' + 'pose'
-        pipe.hmset(key, pose)
+      for target in self.targets.keys():
+        for p_id, pose in self.targets[target].items():
+          key = str(target)+'/' + p_id + '/' + 'pose'
+          pipe.hmset(key, pose)
       pipe.execute()
+
+  def getSitablePose(self, req : GetPose.Request, res : GetPose.Response):
+    '''
+    @brief Service callback to return the list of target keys.
+    The function retrieves all keys from the Redis database and returns them in the response.
+    @param req: The service request (not used in this case).
+    @param res: The service response to populate with target keys.
+    @return A filled GetPose.Response object containing the list of target keys.
+    '''
+    res = GetPose.Response()
+    self.get_logger().info("Targets request")
+    res = self.getPose(dict = 'sitable_positions', key = req.key)
+    return res
+
+
+  def getTargets(self, req : GetPose.Request, res : GetPose.Response):
+    '''
+    @brief Service callback to return the list of target keys.
+    The function retrieves all keys from the Redis database and returns them in the response.
+    @param req: The service request (not used in this case).
+    @param res: The service response to populate with target keys.
+    @return A filled GetPose.Response object containing the list of target keys.
+    '''
+    res = GetPose.Response()
+    self.get_logger().info("Targets request")
+    res = self.getPose(dict = 'targets', key = req.key)
+    return res
   
-  def getPose(self, req : GetPose.Request, res : GetPose.Response):
+  def getPose(self, dict: str, key: str):
```

Comment:
```suggestion
  def getPose(self, req : GetPose.Request, res : GetPose.Response):
```
The function will now be the service callback.